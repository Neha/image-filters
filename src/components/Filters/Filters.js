import React from "react";
import PropTypes from "prop-types";

const Filters = props => {
  return (
    <section>
      <button
        data-name="blur"
        onClick={e => {
          props.onClick(e.target.attributes["data-name"].value);
        }}
      >
        Blur
      </button>
      <button
        data-name="brightness"
        onClick={e => {
          props.onClick(e.target.attributes["data-name"].value);
        }}
      >
        Brightness
      </button>
      <button
        data-name="invert"
        onClick={e => {
          props.onClick(e.target.attributes["data-name"].value);
        }}
      >
        Invert
      </button>
      <button
        data-name="grayscale"
        onClick={e => {
          props.onClick(e.target.attributes["data-name"].value);
        }}
      >
        Grayscale
      </button>
      <button
        data-name="reset"
        onClick={e => {
          props.onClick(e.target.attributes["data-name"].value);
        }}
      >
        reset
      </button>
    </section>
  );
};

Filters.propTypes = {
  onClick: PropTypes.func
};

export default Filters;
