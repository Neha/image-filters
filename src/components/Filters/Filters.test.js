import React from "react";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import renderer from "react-test-renderer";
configure({ adapter: new Adapter() });

import { shallow } from "enzyme";
import Filters from "./Filters";

describe("Filters", () => {
  it("Filters renders without crashing", () => {
    const wrapper = shallow(<Filters />);
    expect(wrapper.exists()).toBe(true);
  });

  it("Filters Snapshot testing", () => {
    const tree = renderer.create(<Filters />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Filters have button", () => {
    const wrapper = shallow(<Filters />);
    expect(wrapper.find("button").length).not.toBeLessThan(0);
  });

  it("Props", () => {
    const wrapper = shallow(<Filters />);
    const mockData = ["Blur", "red", "invert", "Grayscale", "reset"];

    // mockData.map((val, key) => {
    //     console.log(val);
    //   wrapper.find("button").forEach(node => {
    //     expect(node.text()).toEqual(val);
    //   });
    // });

    // for(let i = 0; i > mockData.length; i++) {
    //     wrapper.find('button').forEach(node => {
    //         console.log(node.text())
    //         expect(node.text()).toEqual(mockData[i]);
    //     })
    // }
  });

  it("Filters have Callback", () => {
    const mockCallBack = jest.fn();

    // const wrapper =  shallow(<Filters onClick={mockCallBack} />);
  
    // wrapper.find('button').at(1).simulate('click',()=> {{tempdata}});
    // expect(mockCallBack.mock.calls.length).not.toBeLessThan(0);
  });
});
