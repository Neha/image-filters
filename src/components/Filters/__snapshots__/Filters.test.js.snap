// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filters Filters Snapshot testing 1`] = `
<section>
  <button
    data-name="blur"
    onClick={[Function]}
  >
    Blur
  </button>
  <button
    data-name="brightness"
    onClick={[Function]}
  >
    Brightness
  </button>
  <button
    data-name="invert"
    onClick={[Function]}
  >
    Invert
  </button>
  <button
    data-name="grayscale"
    onClick={[Function]}
  >
    Grayscale
  </button>
  <button
    data-name="reset"
    onClick={[Function]}
  >
    reset
  </button>
</section>
`;
